libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/targets/native/internal/KotlinNativePlatformDependencies.kt

```diff
@@ -61,7 +61,9 @@ internal fun Project.setupKotlinNativePlatformDependencies() {
 internal fun Project.getNativeDistributionDependencies(target: CommonizerTarget): FileCollection {
     return when (target) {
         is LeafCommonizerTarget -> getOriginalPlatformLibrariesFor(target)
-        is SharedCommonizerTarget -> commonizeNativeDistributionTask?.get()?.getCommonizedPlatformLibrariesFor(target) ?: project.files()
+        is SharedCommonizerTarget -> project.files(
```

Comment:
What do you think about defining another function like 
```
TaskProvider<NativeDistributionCommonizerTask>.getCommonizedPlatformLibrariesFor(target: SharedCommonizerTarget) = project.filesProvider {  ... } 
```

on the TaskProvider?
With this, others will have intuitively use this better version instead!